docs/DeveloperGuide.md

```diff
@@ -1,34 +1,384 @@
+* Initializes the User Interface object, as well as the 3 Storage  objects for `BookmarkList`, `Timetable` and `Planner`.
+
+### User Interface component
+
+![](images/UiClassDiagram.png)
+*<center/> Figure 1.3 Class diagram of User Interface </center> <br/></br>*
+
+**API**:`Ui.java`
+
+The UI component is responsible for all visual output to the User from the app. 
+
+The only class carrying out the component's function is the `Ui` class.
+
+It utilises the `Jansi` library to enable the usage of ANSI escape codes to format the console, allowing the app to output in colour. <br/>
+
+The UI also receives input from the User using a `Scanner` object. It returns the input as a String to the main function. <br/>
+
+In addition, the UI contains the different exception and error messages which can be displayed. When a particular exception is thrown (eg. **UNKNOWN_INPUT**), the corresponding function is called in UI to print out the error message (**printUnknownInputMessage()**). 
+
+
+Its main roles are:
+
+* Receiving user commands and returning it to the Main function.
+* Prints visual output in the console for the User
+
+
+### Parser component
+
+
+
+![]()
+*<center/> Figure 1.4 Class diagram of Parser </center> <br/></br>*
+
+
+The Parser component is responsible for decoding the user's input and telling the Main function which command to execute.
+
+It also contains the **programMode** which indicates which mode the program currently is in.
+
+
+It initializes the different commands according to which mode the program currently is in.
+
+It consists of `Parser` and the Command interface classes.
+
+
+Its main roles are:
+
+* Decoding users commands and returning the correct command to the Main function to be executed
+* Catch errors in users commands and return the appropriate exception to the Main function
+* Storing the mode Zoomaster is in (Bookmark/Timetable/Planner)
+
+The interaction of the Parser component with the Command component is covered in greater detail below. 
+
+### Commands component
+Figures 1.3 to 1.6 below show the class-level diagrams for Parser and Command for each different mode. <br/>
+The diagrams are colour coded as such:
+* Orange -> Global
+* Green -> Bookmark Mode
+* Blue -> Timetable Mode
+* Red -> Planner Mode
+
+![](https://raw.githubusercontent.com/Speedweener/ip/master/docs/images/mode1.png)
+*<center/> Figure 1.3 Class diagram of Commands valid in all modes (Global) </center> <br/></br>*
+![](https://raw.githubusercontent.com/Speedweener/ip/master/docs/images/mode2.png)
+*<center/> Figure 1.4 Class diagram of Commands valid in Bookmark Mode </center> <br/></br>*
+![](https://raw.githubusercontent.com/Speedweener/ip/master/docs/images/mode3.png)
+*<center/> Figure 1.5 Class diagram of Commands valid in Timetable Mode</center> <br/></br>*
+![](https://raw.githubusercontent.com/Speedweener/ip/master/docs/images/mode4.png)
+*<center/> Figure 1.6 Class diagram of Commands valid in Planner Mode</center> <br/></br>*
```

Comment:
Perhaps you could consider increasing the size of the diagram so that the words appear bigger?